tio/bufferedreader.go

```diff
@@ -257,6 +271,55 @@ func (buffer *BufferedReader) parseMLE64() ([]byte, int) {
 	return buffer.extractMessage(int(messageLen), buffer.paramMLE+8)
 }
 
+// messages are separated by regexp
+func (buffer *BufferedReader) parseDelimiterMultiLine() ([]byte, int) {
+	startRegexp := buffer.delimiterRegexp
+	lineDelimiter := []byte("\n")
+
+	hasMessage := false
+	messageStarted := false
+	loopStartIdx := 0
+	endIdx := 0
+
+	for {
+		endIdx = bytes.Index(buffer.data[loopStartIdx:buffer.end], lineDelimiter)
+
+		if endIdx < 0 {
+			if messageStarted == true {
+				hasMessage = true
+				endIdx = loopStartIdx
+			}
+			break
+		}
+
+		endIdx++
+		endIdx += loopStartIdx
+
+		startLoc := startRegexp.FindIndex(buffer.data[loopStartIdx:endIdx])
```

Comment:
I finally got some time to do an in depth review.
What I'm asking myself is - why do you even go line-by-line here?
You could just start looking for the regexp in the complete message chunk, which is just a stream of bytes containing linefeeds. Using FindIndex will ignore the linefeeds anyways and you could even use expressions spanning multiple lines this way.
See https://play.golang.org/p/egztEi8m7am

Also this would drastically simplify this loop.

Reply:
See my comment on the unit test why it might not be a good idea to treat trailing new lines as a valid message.